src/inference/Inference.jl

```diff
@@ -369,6 +370,49 @@ function AbstractMCMC.bundle_samples(
     )
 end
 
+function AbstractMCMC.bundle_samples(
+    rng::AbstractRNG,
+    model::AbstractModel,
+    spl::Sampler,
+    N::Integer,
+    ts::Vector{<:AbstractTransition},
+    chain_type::Type{Vector};
+    discard_adapt::Bool=true,
+    save_state=true,
+    kwargs...
+)
+    return ts
+end
+
+function AbstractMCMC.bundle_samples(
+    rng::AbstractRNG,
+    model::AbstractModel,
+    spl::Sampler,
+    N::Integer,
+    ts::Vector{<:AbstractTransition},
+    chain_type::Type{NamedTuple};
+    discard_adapt::Bool=true,
+    save_state=true,
+    kwargs...
+)
+    nts = Vector{NamedTuple}(undef, N)
```

Comment:
This creates an array with an abstract element type - I guess the different elements are not guaranteed to be of the same type? Maybe it could be useful to return `map(i -> nts[i], 1:N)` instead of `nts` at the end, similar to what is done in DiffEqBase for vectors with abstract element types such as in https://github.com/JuliaDiffEq/DiffEqBase.jl/blob/2386f211a2228bb302a0145c039ffa9ca2d2b450/src/ensemble/basic_ensemble_solve.jl#L180?